### PR TITLE
fix: Adjust min/max for 32f_s32f_convert_8i kernel

### LINKED ISSUE #390: volk_32f_s32f_convert_8i behaviour differs between arm and x64
As detailed in https://github.com/gnuradio/gnuradio/issues/3571 it appears that `volk_32f_s32f_convert_8i` does not work as expected on ARM. The behaviour differs from that experienced in x86_64, and appears to unintentionally truncate negative values to zero.

### PR DESCRIPTION
**Pull request by jdemel**
This kernel used `CHAR_MIN` to determine the minimum value which may be
0. It is expected that this value is -127. Thus, we move to `INT8_MIN`.

Fix #390

### CODE REVIEW

**Comment by drmpeg** [comment]
Tested on armv7 and works good. I also verified that CHAR_MIN = 0 and CHAR_MAX = 255 since char is unsigned on ARM.

For consistency (even though they'll never execute on ARM), should the other instances of CHAR_MIN and CHAR_MAX in the module be changed to INT8_MIN and INT8_MAX?

**Comment by jdemel** [comment]
I'd agree that they need to be updated too. Thanks for pointing this out.

**Review by michaelld — APPROVED** [review]
Please change the comment on line 28 : "char" -> "int". After that, LGTM & let's merge this.

**Comment by michaelld** [comment]
Relevant CI passes. Maintainer approval. Important fix. Merging.

**Pull request merged**
